catanatron/models.py

```diff
@@ -0,0 +1,289 @@
+import random
+from enum import Enum
+
+from catanatron.coordinate_system import (
+    generate_coordinate_system,
+    Direction,
+    add,
+    UNIT_VECTORS,
+)
+
+
+class Color(Enum):
+    RED = "RED"
+    BLUE = "BLUE"
+    ORANGE = "ORANGE"
+    WHITE = "WHITE"
+
+
+class Resource(Enum):
+    WOOD = "WOOD"
+    BRICK = "BRICK"
+    SHEEP = "SHEEP"
+    WHEAT = "WHEAT"
+    ORE = "ORE"
+
+
+# Given a tile, the reference to the node.
+class NodeRef(Enum):
+    NORTH = "NORTH"
+    NORTHEAST = "NORTHEAST"
+    SOUTHEAST = "SOUTHEAST"
+    SOUTH = "SOUTH"
+    SOUTHWEST = "SOUTHWEST"
+    NORTHWEST = "NORTHWEST"
+
+
+class EdgeRef(Enum):
+    EAST = "EAST"
+    SOUTHEAST = "SOUTHEAST"
+    SOUTHWEST = "SOUTHWEST"
+    WEST = "WEST"
+    NORTHWEST = "NORTHWEST"
+    NORTHEAST = "NORTHEAST"
+
+
+class Tile:
```

Comment:
Should Tiles contain an index/coordinate/id ?

Reply:
Yes, I think so!

I am finding the coordinate is often enough. But I'll keep this in mind as soon as I find a use case.